src/user_input.rs

```diff
@@ -240,12 +249,12 @@ impl Iterator for InputModeIter {
     }
 }
 
-impl From<InputButton> for InputMode {
-    fn from(button: InputButton) -> Self {
+impl From<InputKind> for InputMode {
+    fn from(button: InputKind) -> Self {
         match button {
-            InputButton::Gamepad(_) => InputMode::Gamepad,
-            InputButton::Keyboard(_) => InputMode::Keyboard,
-            InputButton::Mouse(_) => InputMode::Mouse,
+            InputKind::GamepadButton(_) | InputKind::GamepadAxis(_) => InputMode::Gamepad,

```

Comment:
I'm not 100% what `InputMode` is meant to indicate. Should we have `GamepadButton` and `GamepadAxis` variants of `InputMode`, or just `Gamepad`?

Reply:
As far as I can tell, `InputMode` is used when you want to take some input and simplify it into where it came from. If that is correct, then you're doing the right thing here, as both the gamepad buttons and axes are parts of a gamepad.

Reply:
Precisely.